lib/src/DateRangePicker/DateRangePickerInput.tsx

```diff
@@ -0,0 +1,157 @@
+import { CurrentlySelectingRangeEndProps } from './RangeTypes';
+import { createDelegatedEventHandler } from '../_helpers/utils';
+
+export const useStyles = makeStyles(
+  theme => ({
+    rangeInputsContainer: {
+      display: 'flex',
+      alignItems: 'center',
+      // ? TBD
+      [theme.breakpoints.down('xs')]: {
+        flexDirection: 'column',
+      },
+    },
+    toLabelDelimiter: {
+      margin: '0 16px',
+      [theme.breakpoints.down('xs')]: {
+        margin: '8px 0',
+      },
+    },
+  }),
+  { name: 'MuiPickersDateRangePickerInput' }
+);
+
+export interface ExportedDateRangePickerInputProps {
+  toText?: React.ReactNode;
+}
+
+export interface DateRangeInputProps
+  extends ExportedDateRangePickerInputProps,
+    CurrentlySelectingRangeEndProps,
+    DateInputProps<RangeInput, DateRange> {
+  startText: React.ReactNode;
+  endText: React.ReactNode;
+}
+
+export const DateRangePickerInput: React.FC<DateRangeInputProps> = ({
+  toText = 'to',
+  rawValue,
+  onChange,
+  onClick,
+  parsedDateValue,
+  id,
+  open,
+  className,
+  containerRef,
+  forwardedRef,
+  currentlySelectingRangeEnd,
+  setCurrentlySelectingRangeEnd,
+  openPicker,
+  onFocus,
+  readOnly,
+  disableOpenPicker,
+  startText,
+  endText,
+  ...other
+}) => {
+  const utils = useUtils();
+  const classes = useStyles();
+  const startRef = React.useRef<HTMLInputElement>(null);
+  const endRef = React.useRef<HTMLInputElement>(null);
+  const [start, end] = parsedDateValue;
+
+  React.useEffect(() => {
+    if (!open) {
+      return;
+    }
+
+    if (currentlySelectingRangeEnd === 'start') {
+      startRef.current?.focus();
+    } else if (currentlySelectingRangeEnd === 'end') {
+      endRef.current?.focus();
+    }
+  }, [currentlySelectingRangeEnd, open]);
+
+  const handleStartChange = (date: MaterialUiPickersDate, inputString?: string) => {
+    if (date === null || utils.isValid(date)) {
+      onChange([date, end], inputString);
+    }
+  };
+
+  const handleEndChange = (date: MaterialUiPickersDate, inputString?: string) => {
+    if (date === null || utils.isValid(date)) {
+      onChange([start, date], inputString);
+    }
+  };
+
+  const openRangeStartSelection = () => {
+    if (setCurrentlySelectingRangeEnd) {
+      setCurrentlySelectingRangeEnd('start');
+    }
+    if (!disableOpenPicker) {
+      openPicker();
+    }
+  };
+
+  const openRangeEndSelection = () => {
+    if (setCurrentlySelectingRangeEnd) {
+      setCurrentlySelectingRangeEnd('end');
+    }
+    if (!disableOpenPicker) {
+      openPicker();
+    }
+  };
+
+  return (
+    <div id={id} className={clsx(classes.rangeInputsContainer, className)} ref={containerRef}>
+      <KeyboardDateInput
+        {...other}
+        open={open}
+        forwardedRef={startRef}
+        rawValue={start}
+        parsedDateValue={start}
+        onChange={handleStartChange}
+        disableOpenPicker
+        openPicker={() => {}}
```

Comment:
Dead code?

Reply:
No, it means that we are doing nothing on this step. Probably need to make a function `doNothing` to avoid creating a new function

Reply:
What do you think of making `openPicker` optional? Why having it required? Don't we have here an example that it's not always the expected behavior?

Reply:
Actually from the strong typed system perspective. 
This prop is required. But in this particular context — handled on the level higher (and it is expected, because we are doing wrapper above 2 inputs that changing their behavior).